pkg/resource/notebook_instance_lifecycle_config/custom_delta.go

```diff
@@ -0,0 +1,72 @@
+package notebook_instance_lifecycle_config
+
+import (
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+)
+
+func modifyDeltaCreate(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) *ackcompare.Delta {
+	if ackcompare.HasNilDifference(a.ko.Spec.OnCreate, b.ko.Spec.OnCreate) {
+		delta.Add("Spec.OnCreate", a.ko.Spec.OnCreate, b.ko.Spec.OnCreate)
+		return delta
+	}
+	//check length
+	if a.ko.Spec.OnCreate != nil && b.ko.Spec.OnCreate != nil {
```

Comment:
for interfaces like map, array slice etc, `x==nil` check is not sufficient.
Read more here: https://mangatmodi.medium.com/go-check-nil-interface-the-right-way-d142776edef1